### DISCUSSION TITLE
Versioned docs

### DISCUSSION THREAD

**Discussion by jtraub** [litestar-org/litestar#649] [Ideas]
Starlite evolves at a quite fast pace. The problem here is that Starlite version in user project can lag multiple releases behind while documentation at https://starlite-api.github.io/starlite/ is being regenerated on each commit to the main branch. This might be the source of confusion when the docs say about some new feature that is not available in a version that is available to the user.

Should we start docs versioning so user can view docs about their version? Material for MkDocs supports it
https://squidfunk.github.io/mkdocs-material/setup/setting-up-versioning/

The problem here seems to be repository size and Github Pages limits. I have no idea about docs size nor about Github limits but I think we can work something out. Even keeping docs for last 10 versions looks like an improvement.

**Comment by Goldziher** [comment]
Absolutely. It just painful to setup, but its required I think.

**Reply by provinzkraut** [reply]
Totally agree, it's required.
